### ISSUE TITLE
Remove snapshots and unify them with shape log instead

### ISSUE THREAD

**Issue by kevin-dp** [electric-sql/electric#1741]
Currently, Electric stores an initial snapshot and a shape log for every shape.
However, the snapshot is a sequence of inserts and thus can be stored in the shape log, no need for a separate snapshot.

For example, assume the following initial snapshot:
`[ {op: "insert", row: <row1>}, ..., {op: "insert", row: <row g>} ]`

And shape log:
`[ {op: "insert", row: <row h>}, ..., {op: "insert", row: <row j>} ]`

We can remove the snapshot and unify it with the shape log:
`[ {op: "insert", row: <row1>}, ..., {op: "insert", row: <row g>},  {op: "insert", row: <row h>}, ..., {op: "insert", row: <row j>}  ]`

**Comment by msfstef** [comment]
Worth mentioning that we also need to decide on the addressability of the snapshot entries - currently all of the snapshot is being served ending at offset `0_0` - I suggest snapshot entries are addressed by `0_{insert_index}`, which should still be a a "reserved" offset space.

**Comment by kevin-dp** [comment]
Yep, was thinking the same, makes sense to index the changes inside the initial snapshot.
